### PR TITLE
Add cip-test ExternalSecrets to k8s-infra-prow-build cluster

### PR DESCRIPTION
**Pull request by xmudrii**
This PR adds two ExternalSecrets to the `k8s-infra-prow-build` cluster:

- `k8s-cip-test-prod-service-account` (ServiceAccount for the `k8s-cip-test-prod` GCP project)
- `k8s-gcr-audit-test-prod-service-account` (ServiceAccount for the `k8s-gcr-audit-test-prod` GCP project)

These two GCP projects appear to contain only resources used for running tests and no other production-related resources. At the moment, we have these ExternalSecrets in the `k8s-infra-prow-build-trusted` cluster, however, we can't run presubmit jobs in that cluster (see https://github.com/kubernetes/test-infra/pull/32749). Given that I didn't find any production-related resource in these accounts, I believe it is safe to add these two ExternalSecrets to the non-trusted cluster.

~However, I'm not sure if this is going to work out of the box because I don't know if ESO in the non-trusted cluster can access the `kubernetes-public` project.~ Checked with @upodroid and this should work.

/assign @saschagrunert @cpanato @puerco @Verolop @upodroid @BenTheElder @dims 
cc @kubernetes/release-engineering

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
to get approval both from SIG K8s Infra and SIG Release

**Review by upodroid — APPROVED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
